types/esquery/index.d.ts

```diff
@@ -14,19 +15,35 @@ export = query;
 declare function query(ast: Node, selector: string): Node[];
 
 declare namespace query {
+    interface ESQueryOptions {
+        nodeTypeKey?: string;
+        visitorKeys?: { [nodeType: string]: readonly string[] };
+        fallback?: (node: Node) => string[];
+        matchClass?: (className: string, node: Node, ancestry: Node[]) => boolean;
+    }
+
     /** Parse a selector and return its AST. */
     function parse(selector: string): Selector;
     /** From a JS AST and a selector AST, collect all JS AST nodes that match the selector. */
-    function match(ast: Node, selector: Selector): Node[];
+    function match(ast: Node, selector: Selector, options?: ESQueryOptions): Node[];
     /** Given a `node` and its ancestors, determine if `node` is matched by `selector`. */
-    function matches(node: Node, selector: Selector, ancestry: Node[]): boolean;
+    function matches(node: Node, selector: Selector, ancestry?: Node[] | null, options?: ESQueryOptions): boolean;
```

Comment:
Why did you add the `null` here? I think just having `ancestry?: Node[]` is fine.

Although technically, the function could accept `null`:

https://github.com/estools/esquery/blob/7d1f6691d8f76a85a84214ce210427d6b055b196/esquery.js#L377

```js
    if (!ancestry) { ancestry = []; }
```

I’m not entirely sure what the guidelines about this are but it seems like to me `ancestry` is an optional parameter and I’m more inclined to type it as `ancestry?: Node[]` without the `null`.

Reply:
Besides the fact I was using it in my own code, I've been kind of a mindset that `null` should be allowed where it works and is reasonable, being both more succinct, and also given its meaning that something is consciously set to a nullish value. So I've historically hoped it should be the norm to accept `null` in such cases. However, given the extra typing it requires, perhaps it is not a convention which is likely to win out, and I'm fine dropping it.

Reply:
I've dropped it and gotten the tests passing. Thanks!